tests/lib/nested.sh

```diff
@@ -265,111 +265,184 @@ cleanup_nested_env(){
     rm -rf "$WORK_DIR"
 }
 
-create_nested_core_vm(){
-    # shellcheck source=tests/lib/prepare.sh
-    . "$TESTSLIB"/prepare.sh
+get_image_dir(){
+    echo "$WORK_DIR/image"
+}
 
-    # shellcheck source=tests/lib/snaps.sh
-    . "$TESTSLIB"/snaps.sh
+get_image_name(){
+    local TYPE="$1"
+    local SOURCE="${CORE_CHANNEL}"
+    local VERSION="16"
 
-    mkdir -p "$WORK_DIR/image"
-    if [ ! -f "$WORK_DIR/image/ubuntu-core.img" ]; then
-        local UBUNTU_IMAGE
-        UBUNTU_IMAGE=/snap/bin/ubuntu-image
+    if is_core_20_nested_system; then
+        VERSION="20"
+    elif is_core_18_nested_system; then
+        VERSION="18"
+    fi
 
-        # create ubuntu-core image
-        local EXTRA_FUNDAMENTAL=""
-        local EXTRA_SNAPS=""
-        if [ -d "${PWD}/extra-snaps" ]; then
-            while IFS= read -r mysnap; do
-                EXTRA_SNAPS="$EXTRA_SNAPS --snap $mysnap"
-            done <   <(find extra-snaps -name '*.snap')
-        fi
+    if [ "$BUILD_SNAPD_FROM_CURRENT" = "true" ]; then
+        SOURCE="custom"
+    fi
+    if [ "$(get_extra_snaps | wc -l)" != "0" ]; then
+        SOURCE="custom"
+    fi
+    echo "ubuntu-${TYPE}-${VERSION}-${SOURCE}.img"
+}
+
+get_image_path(){
+    echo "$WORK_DIR/image"
+}
+
+prepare_image_dir(){
+    mkdir -p "$(get_image_dir)"
+}
 
-        local NESTED_MODEL=""
-        case "$SPREAD_SYSTEM" in
+get_extra_snaps(){
+    local EXTRA_SNAPS=""
+    local EXTRA_SNAPS_PATH
+    EXTRA_SNAPS_PATH="$(get_extra_snaps_path)"
+
+    if [ -d "$EXTRA_SNAPS_PATH" ]; then
+        while IFS= read -r mysnap; do
+            echo "$mysnap"
+        done <   <(find "$EXTRA_SNAPS_PATH" -name '*.snap')
```

Comment:
unnecessary whitespace, but feel free to apply in a followup
```suggestion
        done < <(find "$EXTRA_SNAPS_PATH" -name '*.snap')
```